astropy/stats/sigma_clipping.py

```diff
@@ -879,6 +881,287 @@ def sigma_clip(
+            raise ValueError("input data is all masked")
+
+        self.data = sigclip(
+            data, axis=axis, masked=False, return_bounds=False, copy=True
+        )
+        self.axis = axis
+
+    def mean(self) -> float | NDArray:
+        """
+        Calculate the mean of the data.
+
+        Returns
+        -------
+        mean : float or `~numpy.ndarray`
+            The mean of the data.
+        """
+        return nanmean(self.data, axis=self.axis)
+
+    def median(self) -> float | NDArray:
+        """
+        Calculate the median of the data.
+
+        Returns
+        -------
+        median : float or `~numpy.ndarray`
+            The median of the data.
+        """
+        return nanmedian(self.data, axis=self.axis)
+
+    def mode(
+        self,
+        median_factor: float | None = 3.0,
+        mean_factor: float | None = 2.0,
+    ) -> float | NDArray:
+        """
+        Calculate the mode of the data using a estimator of the form
+        ``(median_factor * median) - (mean_factor * mean)``.
+
+        Parameters
+        ----------
+        median_factor : float, optional
+            The multiplicative factor for the data median. Defaults to 3.
+
+        mean_factor : float, optional
+            The multiplicative factor for the data mean. Defaults to 2.
+
+        Returns
+        -------
+        mode : float or `~numpy.ndarray`
+            The estimated mode of the data.
+        """
+        return (median_factor * self.median()) - (mean_factor * self.mean())
+
+    def std(
+        self,
+        ddof: int | None = 0,
+    ) -> float | NDArray:
+        """
+        Calculate the standard deviation of the data.
+
+        Parameters
+        ----------
+        ddof : int, optional
+            The delta degrees of freedom. The divisor used in the
+            calculation is ``N - ddof``, where ``N`` represents the
+            number of elements. The default is 0.
+
+        Returns
+        -------
+        std : float or `~numpy.ndarray`
+            The standard deviation of the data.
+        """
+        return nanstd(self.data, axis=self.axis, ddof=ddof)
+
+    def var(
+        self,
+        ddof: int | None = 0,
+    ) -> float | NDArray:
+        """
+        Calculate the variance of the data.
+
+        Parameters
+        ----------
+        ddof : int, optional
+            The delta degrees of freedom. The divisor used in the
+            calculation is ``N - ddof``, where ``N`` represents the
+            number of elements. The default is 0.
+
+        Returns
+        -------
+        var : float or `~numpy.ndarray`
+            The variance of the data.
+        """
+        return nanvar(self.data, axis=self.axis, ddof=ddof)
+
+    def biweight_location(
+        self,
+        c: float | None = 6.0,
+        M: float | None = None,
+    ) -> float | NDArray:
+        """
+        Calculate the biweight location of the data.
+
+        NaN values are ignored.
```

Comment:
Should this note be added to `var`, `std`, `mode`, `median` and `mean`'s respective docstrings too ?

Reply:
fixed